shared/test/test_animations.rb

```diff
@@ -0,0 +1,359 @@
+  end
+
+  def list_versions(filename)
+    response = @files_api.get "/v3/animations/#{@channel_id}/#{filename}/versions"
+    JSON.parse(response.body)
+  end
+
+  def get(filename, body = '', headers = {})
+    @files_api.get "/v3/animations/#{@channel_id}/#{filename}", body, headers
+    @files_api.last_response.body
+  end
+
+  def get_version(filename, version_id)
+    @files_api.get("/v3/animations/#{@channel_id}/#{filename}?version=#{version_id}").body
+  end
+
+  def delete(filename)
+    @files_api.delete "/v3/animations/#{@channel_id}/#{filename}"
+  end
+
+  def upload(filename, contents, content_type)
+    upload_version(filename, nil, contents, content_type)
+  end
+
+  def upload_version(filename, version_id, contents, content_type)
+    url = "/v3/animations/#{@channel_id}/#{filename}"
+    query_params = version_id.nil? ? '' : "?version=#{version_id}"
+
+    body = { files: [create_uploaded_file(filename, contents, content_type)] }
+    @files_api.post(url + query_params, body, 'CONTENT_TYPE' => content_type)
+    @files_api.last_response.body
+  end
+
+  def create_uploaded_file(filename, contents, content_type)
+    Dir.mktmpdir do |dir|
+      file_path = "#{dir}/#{filename}"
+      File.open(file_path, 'w') do |file|
+        file.write(contents)
+        file.rewind
+      end
+      Rack::Test::UploadedFile.new(file_path, content_type)
+    end
+  end
+
+  def copy(source_filename, dest_filename)
+    @files_api.put("/v3/animations/#{@channel_id}/#{dest_filename}?src=#{CGI.escape(source_filename)}")
+    @files_api.last_response.body
+  end
+
+  # Delete all versions of the specified file from S3, including all delete markers
+  def delete_all_versions(bucket, key)
+    s3 = Aws::S3::Client.new
+    response = s3.list_object_versions(bucket: bucket, prefix: key)
+    objects = response.versions.concat(response.delete_markers).map do |version|
+      {
+          key: key,
+          version_id: version.version_id
+      }
+    end
+    s3.delete_objects(
+        bucket: bucket,
+        delete: {
+            objects: objects,
+            quiet: true
+        }
+    ) if objects.any?
+  end
+
+  def delete_all_animation_versions(filename)
+    delete_all_versions(CDO.animations_s3_bucket, "animations_test/1/1/#{filename}")
+  end
+
+  def successful?
+    @files_api.last_response.successful?
+  end
+
+  def not_found?
+    @files_api.last_response.not_found?
+  end
+
+  def unsupported_media_type?
+    @files_api.last_response.status == 415
+  end
+
+  def assert_fileinfo_equal(expected, actual)
+    assert_equal(Hash, actual.class)
+    assert_equal(expected['filename'], actual['filename'])
+    assert_equal(expected['category'], actual['category'])
+    assert_equal(expected['size'], actual['size'])
+  end
+
+  def randomize_filename(filename)
+    basename = [filename.split('.')[0], '.' + filename.split('.')[1]]
+    basename[0] + '_' + @random.bytes(10).unpack('H*')[0] + basename[1]
+  end
```

Comment:
It seems like whole bunch of duplicated code could be eliminated from these helper functions by having these 3 tests inherit from a common FilesApiTest(?) subclass. Would you consider getting one started, even if only some of the helpers can be easily moved there at this time?